rest/data/project_settings.go

```diff
@@ -15,18 +17,48 @@ import (
 // This file is used to combine operations across data connectors, to avoid
 // duplicated connector usage across the codebase.
 
-func (sc *DBConnector) CopyProject(ctx context.Context, projectToCopy *model.ProjectRef, newProject string) (*restModel.APIProjectRef, error) {
-	// copy project, disable necessary settings
+type CopyProjectOpts struct {
+	ProjectIdToCopy      string
+	NewProjectIdentifier string
+	NewProjectId         string
+}
+
+func (sc *DBConnector) CopyProject(ctx context.Context, opts CopyProjectOpts) (*restModel.APIProjectRef, error) {
+	projectToCopy, err := sc.FindProjectById(opts.ProjectIdToCopy, false)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Database error finding project '%s'", opts.ProjectIdToCopy)
+	}
+	if projectToCopy == nil {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("project '%s' doesn't exist", opts.ProjectIdToCopy),
+		}
+	}
+
+	// verify project with new name doesn't exist
+	if err = sc.verifyUniqueProject(opts.NewProjectIdentifier); err != nil {
+		return nil, err
+	}
+
 	oldId := projectToCopy.Id
+	if opts.NewProjectId != "" {
+		if err = sc.verifyUniqueProject(opts.NewProjectId); err != nil {
```

Comment:
I agree that we should add this to CreateProject as well. We can also do it in a separate ticket if you prefer that.